From 8164f12df6412b35e28ee24bd06d4da975590c9b Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sat, 4 May 2013 13:52:21 +0200
Subject: [PATCH 420/482] 	* grub-core/normal/menu_text.c
 (menu_clear_timeout): Clear second 	line of timeout as it may contain the
 rest of long line.

---
 ChangeLog                    | 5 +++++
 grub-core/normal/menu_text.c | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 16456f0..8d41517 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-05-04  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* grub-core/normal/menu_text.c (menu_clear_timeout): Clear second
+	line of timeout as it may contain the rest of long line.
+
+2013-05-04  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	* grub-core/normal/main.c: Fix freed memory dereference.
 
 2013-05-04  Vladimir Serbinenko  <phcoder@gmail.com>
diff --git a/grub-core/normal/menu_text.c b/grub-core/normal/menu_text.c
index 19a5389..d7e5641 100644
--- a/grub-core/normal/menu_text.c
+++ b/grub-core/normal/menu_text.c
@@ -460,6 +460,8 @@ menu_text_clear_timeout (void *dataptr)
 
   grub_term_gotoxy (data->term, 0, grub_term_height (data->term) - 3);
   grub_print_spaces (data->term, grub_term_width (data->term) - 1);
+  grub_term_gotoxy (data->term, 0, grub_term_height (data->term) - 2);
+  grub_print_spaces (data->term, grub_term_width (data->term) - 1);
   grub_term_gotoxy (data->term, grub_term_cursor_x (data->term),
 		    GRUB_TERM_FIRST_ENTRY_Y + data->offset);
   grub_term_refresh (data->term);
-- 
1.8.2.1

